### DISCUSSION TITLE
MCTS search

### DISCUSSION THREAD

**Discussion by shlosmem** [suragnair/alpha-zero-general#150] [Conceptual]
line 115 in MCTS.p: next_s, next_player = self.game.getNextState(canonicalBoard, 1, a)
Why the player is a constant 1?

**Comment by JernejHabjan** [comment]
I think canonical board gets inverted, so player -1 becomes 1 and other way around, so we are checking for new player (1), who was previously player -1

**Comment by shlosmem** [comment]
so we must assume symmetry in the game rules between the 2 players.

**Comment by JernejHabjan** [comment]
symmetry only in the sense of name for pieces -> -1 for player -1 and 1 for player 1

**Comment by mikhail** [comment]
I'm also running into a case where inverted board is inconvenient. Is there a fault, or a downside to skipping the entire canonical-board rotation, and calling MCTS search as player-aware?
